operator/pkg/utils/iamauthenticator/awsiamauthenticator.go

```diff
@@ -0,0 +1,215 @@
+		InitContainers: []v1.Container{{
+			Name:  "chown",
+			Image: imageprovider.BusyBox(),
+			Command: []string{"sh", "-c",
+				"touch /var/aws-iam-authenticator/kubeconfig/kubeconfig.yaml && chown -R 10000:10000 /var/aws-iam-authenticator/state/ && chown -R 10000:10000 /var/aws-iam-authenticator/kubeconfig && ls -lrt /var/aws-iam-authenticator",
+			},
+			SecurityContext: &v1.SecurityContext{AllowPrivilegeEscalation: ptr.Bool(true)},
+			VolumeMounts: []v1.VolumeMount{{
+				Name:      "state",
+				MountPath: "/var/aws-iam-authenticator/state/",
+			}, {
+				Name:      "kubeconfig",
+				MountPath: "/var/aws-iam-authenticator/kubeconfig/",
+			}},
+		}},
+		Containers: []v1.Container{{
+			Name:  "aws-iam-authenticator",
+			Image: imageprovider.AWSIamAuthenticator(),
+			Args: []string{
+				"server",
+				"--master=https://localhost/",
+				"--config=/etc/aws-iam-authenticator/config.yaml",
+				"--state-dir=/var/aws-iam-authenticator/state/",
+				"--generate-kubeconfig=/var/aws-iam-authenticator/kubeconfig/kubeconfig.yaml",
+			},
+			SecurityContext: &v1.SecurityContext{AllowPrivilegeEscalation: ptr.Bool(true)},
+			VolumeMounts: []v1.VolumeMount{{
+				Name:      "config",
+				MountPath: "/etc/aws-iam-authenticator/",
+			}, {
+				Name:      "state",
+				MountPath: "/var/aws-iam-authenticator/state/",
+			}, {
+				Name:      "kubeconfig",
+				MountPath: "/var/aws-iam-authenticator/kubeconfig/",
+			}},
+		}},
+		Volumes: []v1.Volume{{
+			Name: "kubeconfig",
+			VolumeSource: v1.VolumeSource{
+				HostPath: &v1.HostPathVolumeSource{
+					Path: "/var/aws-iam-authenticator/kubeconfig/",
+				},
+			},
+		}, {
+			Name: "state",
+			VolumeSource: v1.VolumeSource{
+				HostPath: &v1.HostPathVolumeSource{
+					Path: "/var/aws-iam-authenticator/state/",
+				},
+			},
+		}},
+	}
+}
+
+func authenticatorLabels() map[string]string {
+	return map[string]string{"component": "aws-iam-authenticator"}
+}
+
+var (
+	authenticatorConfig = `
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ .Name }}
+  namespace: {{ .Namespace }}
+data:
+  config.yaml: |
+    clusterID: {{ .ClusterName }}
+    server:
+      mapRoles:
+        - groups:
+          - system:bootstrappers
+          - system:nodes
+          rolearn: arn:aws:iam::{{ .AWSAccountID }}:role/{{ .KitNodeRole }}
+          username: system:node:{{ .PrivateDNS}}
+      # List of Account IDs to whitelist for authentication
+      mapAccounts:
+        - {{ .AWSAccountID }}
+`
+)
+
+// TODO move this to util. ParseTemplate validates and parses passed as argument template
+func ParseTemplate(strtmpl string, obj interface{}) ([]byte, error) {
+	var buf bytes.Buffer
+	tmpl := template.Must(template.New("Text").Parse(strtmpl))
+	err := tmpl.Execute(&buf, obj)
+	if err != nil {
+		return nil, fmt.Errorf("error when executing template, %w", err)
+	}
+	return buf.Bytes(), nil
+}
+
+func AuthenticatorDaemonSetName(clusterName string) string {
```

Comment:
You're already in the authenticator package, no need to stutter on naming. Any reason we can't unify the names of all these things? "cluster:component" or "etarndev:authenticator"